src/argilla/client/feedback/dataset/remote.py

```diff
@@ -0,0 +1,247 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional, Union
+
+from tqdm import trange
+
+from argilla.client.feedback.constants import FETCHING_BATCH_SIZE, PUSHING_BATCH_SIZE
+from argilla.client.feedback.dataset.base import FeedbackDatasetBase
+from argilla.client.feedback.schemas.records import FeedbackRecord, RemoteFeedbackRecord
+from argilla.client.sdk.v1.datasets import api as datasets_api_v1
+
+if TYPE_CHECKING:
+    from uuid import UUID
+
+    import httpx
+
+    from argilla.client.feedback.dataset.local import FeedbackDataset
+    from argilla.client.feedback.types import AllowedFieldTypes, AllowedQuestionTypes
+    from argilla.client.sdk.v1.datasets.models import FeedbackItemModel
+    from argilla.client.workspaces import Workspace
+
+
+warnings.simplefilter("always", DeprecationWarning)
+
+
+class RemoteFeedbackRecords:
+    def __init__(self, client: "httpx.Client", id: "UUID", questions: List["AllowedQuestionTypes"]) -> None:
+        self.client = client
+        self.id = id
+
+        self.__question_id2name = {question.id: question.name for question in questions}
+        self.__question_name2id = {value: key for key, value in self.__question_id2name.items()}
+
+    def __repr__(self) -> str:
+        return (
+            "The `records` of a `FeedbackDataset` in Argilla are being lazily"
+            " fetched, and never stored locally. You can either loop over `records`"
+            " or access them by index, and those will be fetched from Argilla on the"
+            " fly."
+        )
```

Comment:
I think it's a bit weird that if you print `ArgillaFeedbackDataset.records` you get this, it's more like a warning message. This method could return something like:

`[RemoteFeedbackRecord(...), RemoteFeedbackRecord(...), ...] lazily fetched from 'https://dev.argilla.io'`

Reply:
Yes, I agree with @gabrielmbmb. Maybe we can just print some of the records:
```suggestion
        return f"{self.records[:3]!r}..."
```

Reply:
Even if I'm not much comfortable making remote calls in `__repr__` method. But can be useful to users.

Reply:
I think this is not optimal and can also be confusing 😩 Maybe we can just go with @gabrielmbmb's suggestion to show something like `[RemoteFeedbackRecord(...), RemoteFeedbackRecord(...), ...] lazily fetched from 'https://dev.argilla.io'`?

Reply:
Or maybe just return `None` and a `warnings.warn` instead